spec/jobs/searchgov_domain_destroyer_job_spec.rb

```diff
@@ -1,38 +1,67 @@
 require 'spec_helper'
 
 describe SearchgovDomainDestroyerJob do
-  subject(:perform) { described_class.perform_now(**args) }
+  subject(:perform) { described_class.perform_now(searchgov_domain) }
 
-  let(:domain) { 'www.archive.gov' }
-  let!(:searchgov_domain) {
-    SearchgovDomain.create!(domain: domain, status: 200)
-  }
-  let(:args) {
-    { searchgov_domain: searchgov_domain }
-  }
+  let(:searchgov_domain) { SearchgovDomain.create(domain: 'www.archive.gov', status: '200 OK') }
+  let(:error_messages) { [] }
+
+  before do
+    allow(Rails.logger).to receive(:error) { |message| error_messages << message }
+  end
 
   describe '#perform' do
     it 'requires a searchgov_domain as an argument' do
-      expect{ described_class.perform_now }.
-        to raise_error(ArgumentError, /missing keyword: :?searchgov_domain/)
+      expect { described_class.perform_now }.to raise_error(ArgumentError)
     end
 
     it 'destroys the searchgov domain' do
-      expect(searchgov_domain).to receive(:destroy!)
       perform
+      expect(SearchgovDomain.exists?(searchgov_domain.id)).to be false
     end
 
     context 'when the searchgov_domain has associated searchgov_url records' do
-      let(:url1) { 'https://www.archive.gov/info' }
-      let(:url2) { 'https://www.archive.gov/hmmm' }
-      let!(:searchgov_url1) { SearchgovUrl.create!(url: url1) }
-      let!(:searchgov_url2) { SearchgovUrl.create!(url: url2) }
+      let!(:searchgov_url1) { searchgov_domain.searchgov_urls.create(url: 'https://www.archive.gov/info', hashed_url: 'hash1') }
+      let!(:searchgov_url2) { searchgov_domain.searchgov_urls.create(url: 'https://www.archive.gov/hmmm', hashed_url: 'hash2') }
 
-      it 'destroys the searchgov_urls' do
-        expect { perform }.to change{ SearchgovUrl.count }.by(-2)
+      it 'destroys all associated searchgov_urls' do
+        perform
+        expect(SearchgovUrl.exists?(searchgov_url1.id)).to be false
+        expect(SearchgovUrl.exists?(searchgov_url2.id)).to be false
+      end
+    end
+
+    context 'when there are errors' do
```

Comment:
We haven't have "dangling" undeleted urls. This is not an issue and not something we need to worry about right now. Just let delete do it's thing and trust the framework in reporting back any error